File: year_2022/src/aoc/Day1.scala
package aoc

import scala.io.Source

object Day1 {

  def main(args: Array[String]): Unit = {
    val t0: Long = System.nanoTime()
    val source: Source = Source.fromFile(s"$inputPrefix/day1.txt")
    val input = source.getLines().mkString("\n")

    println(input)
    println(part2(input))

    println("Elapsed time: " + (System.nanoTime() - t0) / 1000000 + " ms")
    source.close()
  }

  def part1(lines: String): Int = {
    val l = lines.split("\n\n")
    l.map(i => {
      println("group\n" + i )
      val sum = i.split("\n").map(s => s.toInt).sum
      println(s"sum: $sum")
      sum
    }).max
  }

  def part2(lines: String): Int = {
    val l = lines.split("\n\n")
    l.map(i => {
      println("group\n" + i)
      val sum = i.split("\n").map(s => s.toInt).sum
      println(s"sum: $sum")
      sum
    }).sorted.takeRight(3).sum
  }
}

File: year_2022/src/aoc/Day1.py

def parseInput(day):
    path = __file__.rstrip(f"Day{day}.py") + f"../../input/day{day}.txt"
    print("Input file:", path)
    with open(path, 'r') as file:
        lines = [line.strip() for line in file]
        groups = []
        currGroup = 0
        for line in lines:
            if line == "":
                groups.append(currGroup)
                currGroup = 0
            else:
                currGroup += int(line)
        return groups

# def part1():
#     True
#
# def part2():
#     lines = parseInput(1)

if __name__ == "__main__":
    print("\nPART 1 RESULT")
    lines = parseInput(1)
    print(max(lines))

    print("\nPART 2 RESULT")
    print(sum(sorted(lines,reverse=True)[0:3]))
